gotools/controller-gen/go.mod

```diff
@@ -0,0 +1,41 @@
+module github.com/rancher/rancher/gotools/controller-gen
+
+go 1.24.2
```

Comment:
Although not required but it would be better to keep `go` and `toolchain` directives in sync with the main `go.mod` file.

Reply:
Updated. This made me think a bit about the Go and how those tools are used..

Two things I haven't realized with the go tool approach:
1. We're building those binaries with different Go version than their release. That can be fine, eg: mockgen's install instruction is to use `go install` which is pretty much what `go tool` is doing if you were using GOTOOLCHAIN=local.
2. If a tool doesn't compile with a latest Go toolchain, it's going to prevent us from updating that toolchain (until a fix is pushed upstream or we change the way we build the tool).

I think we'll be fine though, not sure it's worth abandoning this PR for this.